src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala

```diff
@@ -0,0 +1,175 @@
+    geneAssociation => (geneAssociation \ "DisorderGeneAssociationStatus" \ "Name").headOption.map(_.text)
+
+  def parseProduct6XML(doc: Elem): Seq[Product6] = {
+    for {
+      disorder <- doc \\ "DisorderList" \\ "Disorder"
+      orphaNumber <- disorder \ "OrphaCode"
+      expertLink <- disorder \ "ExpertLink"
+      name <- disorder \ "Name"
+      disorderType <- disorder \ "DisorderType"
+      disorderTypeName <- disorderType \ "Name"
+      disorderGroup <- disorder \ "DisorderGroup"
+      disorderGroupName <- disorderGroup \ "Name"
+      geneAssociation <- disorder \ "DisorderGeneAssociationList" \ "DisorderGeneAssociation"
+      genes <- geneAssociation \ "Gene"
+      locus <- genes \ "LocusList" \ "Locus"
+    } yield {
+      Product6(
+        disorder.attribute("id").get.text.toLong,
+        orphaNumber.text.toLong,
+        expertLink.text,
+        name.text,
+        disorderType.attribute("id").get.text.toLong,
+        disorderTypeName.text,
+        disorderGroup.attribute("id").get.text.toLong,
+        disorderGroupName.text,
+        (geneAssociation \ "SourceOfValidation").text,
+        genes.attribute("id").get.text.toLong,
+        (genes \ "Symbol").text,
+        (genes \ "Name").text,
+        (genes \ "SynonymList" \\ "Synonym").map(_.text).toList,
+        ensembl_gene_id(genes),
+        genatlas_gene_id(genes),
+        HGNC_gene_id(genes),
+        omim_gene_id(genes),
+        reactome_gene_id(genes),
+        swiss_prot_gene_id(genes),
+        association_type(geneAssociation),
+        association_type_id(geneAssociation),
+        association_status(geneAssociation),
+        locus.attribute("id").get.text.toLong,
+        (genes \ "LocusList" \ "Locus" \ "GeneLocus").text,
+        (genes \ "LocusList" \ "Locus" \ "LocusKey").text.toLong
+      )
+    }
+  }
+
+  def parseProduct9XML(doc: Elem): Seq[Product9] = {
+    for {
+      disorder <- doc \\ "DisorderList" \\ "Disorder"
+      orphaCode <- disorder \ "OrphaCode"
+      expertLink <- disorder \ "ExpertLink"
+      name <- disorder \ "Name"
+      disorderType <- disorder \ "DisorderType"
+      disorderTypeName <- disorderType \ "Name"
+      disorderGroup <- disorder \ "DisorderGroup"
+      disorderGroupName <- disorderType \ "Name"
+    } yield {
+      Product9(
+        disorder.attribute("id").get.text.toLong,
+        orphaCode.text.toLong,
+        expertLink.text,
+        name.text,
+        disorderType.attribute("id").get.text.toLong,
+        disorderTypeName.text,
+        disorderGroup.attribute("id").get.text.toLong,
+        disorderGroupName.text,
+        (disorder \\ "AverageAgeOfOnsetList" \ "AverageAgeOfOnset" \ "Name").map(_.text).toList,
+        (disorder \\ "AverageAgeOfDeathList" \ "AverageAgeOfDeath" \ "Name").map(_.text).toList,
+        (disorder \\ "TypeOfInheritanceList" \ "TypeOfInheritance" \ "Name").map(_.text).toList
+      )
+    }
+  }
+
+}
+
+case class Product6(disorder_id: Long,
```

Comment:
I don't know the business logic needed, but the naming is not very telling in my opinion. "Product6" what does it represent?

Reply:
This is the name of the files. I think it's my bad, I sent the link without giving any indications about these files. Product6 === Genes associated with rare diseases
Product9 === Rare diseases natural history

Reply:
I agree, I'll change these